ff/src/fields/models/quadratic_extension.rs

```diff
@@ -192,11 +192,17 @@ impl<P: QuadExtParameters> One for QuadExtField<P> {
 
 impl<P: QuadExtParameters> Field for QuadExtField<P> {
     type BasePrimeField = P::BasePrimeField;
+    type BasePrimeFieldArr = Vec<Self::BasePrimeField>;
```

Comment:
We'll have to explicitly specify the type of the iterator; the one I'm suggesting below might not be the correct type, but is probably not too far from the correct one.
```suggestion

type BaseFieldIter<P> = <P::BaseField as Field>::BasePrimeFieldIter;

impl<P: QuadExtParameters> Field for QuadExtField<P> {
    type BasePrimeField = P::BasePrimeField;
    type BasePrimeFieldArr = Chain<BaseFieldIter<P>, BaseFieldIter<P>>;
```